bodo/pandas/series.py

```diff
@@ -804,11 +855,32 @@ def sig_bind(name, accessor_type, *args, **kwargs):
 }
 
 
-def gen_method(name, return_type, is_method=True, accessor_type=""):
+def validate_dtype(name, obj):
```

Comment:
As above, I'd just validate in constructor of the accessor class and the methods themselves.